### Dataset



In [1]:
import pandas as pd
import numpy as np

In [3]:
dc_listings=pd.read_csv('dc_airbnb.csv')

In [4]:
dc_listings.head()

host_response_rate host_acceptance_rate  host_listings_count  accommodates  \
0                92%                  91%                   26             4   
1                90%                 100%                    1             6   
2                90%                 100%                    2             1   
3               100%                  NaN                    1             2   
4                92%                  67%                    1             4   

         room_type  bedrooms  bathrooms  beds    price cleaning_fee  \
0  Entire home/apt       1.0        1.0   2.0  $160.00      $115.00   
1  Entire home/apt       3.0        3.0   3.0  $350.00      $100.00   
2     Private room       1.0        2.0   1.0   $50.00          NaN   
3     Private room       1.0        1.0   1.0   $95.00          NaN   
4  Entire home/apt       1.0        1.0   1.0   $50.00       $15.00   

  security_deposit  minimum_nights  maximum_nights  number_of_reviews  \
0          $100.00               1            1125                  0   
1              NaN               2              30                 65   
2              NaN               2            1125                  1   
3              NaN               1            1125                  0   
4          $450.00               7            1125                  0   

    latitude  longitude           city zipcode state  
0  38.890046 -77.002808     Washington   20003    DC  
1  38.880413 -76.990485     Washington   20003    DC  
2  38.955291 -76.986006    Hyattsville   20782    MD  
3  38.872134 -77.019639     Washington   20024    DC  
4  38.996382 -77.041541  Silver Spring   20910    MD

### 1.Distance
 Calculate the Euclidean distance between our living space, which can accommodate 3 people,
 and the **_first living space_** in the dc_listings Dataframe.

In [7]:
first_distance=abs(3-dc_listings.iloc[0]["accommodates"])

Calculate the distance between each value in the accommodates column from
dc_listings and the value 3, which is the number of people our listing accommodates:

- Use the apply method to calculate the absolute value between each value in accommodates
and 3 and return a new Series containing the distance values.

In [14]:
dc_listings['distance']=dc_listings["accommodates"].apply(lambda x: abs(3-x))

In [16]:
print(dc_listings['distance'].value_counts())

1     2294
2      503
0      461
3      279
5       73
4       35
7       22
6       17
9       12
13       8
8        7
12       6
11       4
10       2
Name: distance, dtype: int64


### 2. Find the nearest 5
If we sort by the distance column and then just select the first 5 living spaces,
we would be biasing the result to the ordering of the dataset.

Let's instead randomize the ordering of the dataset and then sort the Dataframe by the distance column.
This way, all of the living spaces with the same number of bedrooms will still be at
the top of the Dataframe but will be in random order across the first 461 rows.


In [17]:
np.random.seed(1)
new_id=np.random.permutation(len(dc_listings))
dc_listings=dc_listings.loc[new_id]
dc_listings=dc_listings.sort_values('distance')
print(dc_listings['price'].head(10))

577     $185.00
2166    $180.00
3631    $175.00
71      $128.00
1011    $115.00
380     $219.00
943     $125.00
3107    $250.00
1499     $94.00
625     $150.00
Name: price, dtype: object


### 3. Clean the "Price" column
Remove the commas (,) and dollar sign characters ($) from the price column:
- Use the str accessor so we can apply string methods to each value in the column followed by the string method replace
to replace all comma characters with the empty character:
**stripped_commas = dc_listings['price'].str.replace(',', '')**
- Repeat to remove the dollar sign characters as well.


In [18]:
stripped_commas=dc_listings['price'].str.replace(',','').str.replace('$','')
dc_listings['price']=stripped_commas.astype(float)
mean_price=dc_listings['price'].head(5).mean()
print(mean_price)


156.6


### 4. Function to make prediction
